app/controllers/rx_controller.rb

```diff
@@ -4,13 +4,21 @@
 class RxController < ApplicationController
   include ActionController::Serialization
 
-  skip_before_action :authenticate
+  before_action :authorize_rx
   before_action :authenticate_client
 
   protected
 
   def client
-    @client ||= Rx::Client.new(session: { user_id: ENV['MHV_USER_ID'] })
+    @client ||= Rx::Client.new(session: { user_id: mhv_correlation_id })
+  end
+
+  def authorize_rx
+    mhv_correlation_id || raise(Common::Exceptions::Unauthorized)
```

Comment:
403 it is then. 👍 

@kreek yes, my goal was that we would support minor codes. That's actually what "code" in the rendered json is supposed signify. My thought was that we would create inherited versions of the existing exceptions, for each minor code since it is a distinctly different exception. This way an API consumer has a unique code that maps directly to a "detail" description, "detail" might change, but the minor code will never change.

For anything very granular but still specific to a particular minor level, one could use "source" as well. Checkout the JSONAPI spec for details on this. Naturally this is all open to feedback and improvement based on our needs, but that was sort of the thought I had when I built these exception classes.